agent/engine/task_manager.go

```diff
@@ -350,7 +350,7 @@ func (mtask *managedTask) steadyState() bool {
 	}
 }
 
-// cleanupCredentials removes credentials for a stopped task
+// cleanupCredentials removes credentials for a stopped task (execution credentials are removed in cleanupTask)
```

Comment:
Can we elaborate on why? I'm guessing because we need the execution role to call FE stopTask during `cleanupTask`, but would be good if we make it clear here.

Reply:
The initial aspiration of moving execution role to the later part of clean up process is that unlike container role, execution role might be in use till the last phase of clean up such as sending task cleanup logs to cloudwatch.

As for calling stopTask I don't think it would require executionRole as a task can be run and stopped without an execution role. Correct me if I am wrong :)

Reply:
Regarding the memory leak one the testing setup I had is not very correct, I synced with Feng this morning, I will need to change the ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION and wait for the full clean up which isn't included in the current test. I will do another round of testing to see more accurate result.